runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -384,7 +384,7 @@ public State getState() {
       Collection<PTransform<?, ?>> steps = aggregatorSteps.get(aggregator);
       final Map<String, T> stepValues = new HashMap<>();
       for (AppliedPTransform<?, ?, ?> transform : evaluationContext.getSteps()) {
-        if (steps.contains(transform.getTransform())) {
+        if (steps != null && steps.contains(transform.getTransform())) {
```

Comment:
You may as well skip the entire loop if we know the `Aggregator` is not used within the graph (or at least not declared during graph construction)